### ISSUE TITLE
Multiple concurrent delay & loss NetworkChaos get overridden

### ISSUE THREAD

**Issue by rustielin** [chaos-mesh/chaos-mesh#3631]
## Bug Report

**What version of Kubernetes are you using?**
1.22
**What version of Chaos Mesh are you using?**
Controller manager Version: version.Info{GitVersion:"v2.2.0", GitCommit:"22a1cf57398e35da58dd391b79b64f3071e2863b", BuildDate:"2022-04-29T13:15:01Z", GoVersion:"go1.18", Compiler:"gc", Platform:"linux/amd64"}

**What did you do?**
Applying both `loss` and `delay` types of `NetworkChaos` targeting the same pods. Toggling them manually via Chaos Dashboard.

**What did you expect to see?**
Expecting both the `loss` and `delay` chaoses to stack, and having both effects at the same time. I could not also find documentation around the behavior of multiple concurrent NetworkChaos.

**What did you see instead?**
The last one applied overrides the existing NetworkChaos.

**Output of chaosctl**

**Comment by STRRL** [comment]
another cases with overlapping situation: https://github.com/chaos-mesh/chaos-mesh/issues/3634

**Comment by rustielin** [comment]
Thanks @STRRL .

Is this the intended behavior? Is this on the roadmap somewhere? This is pretty important for anyone who wishes to emulate internet conditions within their setup (delay + loss + bandwidth limit, etc). Thanks!

**Comment by STRRL** [comment]
> Is this the intended behavior? 

Yes, for now, that's expected behavior, but actually does not make so much sense. 🥲

For more detail, we did NOT aggregate multiple various policy before apply the `tc` rules, all the `loss`, `delay`, `bandwidth` and so on are built based on `tc` so that you could see that the post chaos experiment would overwrite the former one.

> Is this on the roadmap somewhere? This is pretty important for anyone who wishes to emulate internet conditions within their setup (delay + loss + bandwidth limit, etc). Thanks!

Yes, after inject 2 chaos experiments with 50 ms network delay, we should observe the 100 ms delay in the target pod. I think it should be considered in the future, but there is not a clear roadmap.

**Comment by rustielin** [comment]
Thanks for the response @STRRL . Can you list out exactly which rules are built off of `tc` and thus will not be able to use multiple of? From my experimentation, I am able to use both `delay` and `bandwidth` at the same time, but perhaps it's due to a race in the effects overwriting each other.

Also it would be great if this was on the website documentation somewhere as a caveat :)

**Comment by STRRL** [comment]
> Can you list out exactly which rules are built off of tc and thus will not be able to use multiple of? 

`loss`, `delay`, `duplicate` and `corrupt` could not be overlapped on the same pod by different `NetworkChaos`.

**I just realized that we could use `action: netem` to inject the combination of `loss` and `delay` in one `NetworkChaos` instead of a `loss` NetworkChaos plus one `delay` NetworkChaos.** I think that could resolve your situation. @rustielin 

> Also it would be great if this was on the website documentation somewhere as a caveat :)

That's great! I would append some messages on the NetworkChaos documentation! Thanks!

**Comment by craig-seeman** [comment]
@STRRL I hope I am not adding too much extra noise to this issue but I'm curious if a possible solution for this problem could be to add more state tracking on the part of the controller-manager and have it 'remember' what all was ran and where? 

This added ability would certainly help with the issues we are having currently with chaos-mesh at scale (new pods do not get experiments assigned to them after an experiment starts, rendering quite a few experiments invalid for us)

Related: #2976

**Comment by STRRL** [comment]
Hi @craig-seeman , sorry for the late response.

Chaos Mesh has restored some information about the injected Pods as `records` in `<Some>Chaos.Status`, but we have not implemented the "dynamic" part. Because the dynamic logic part might be tricky, and there is no plan to make it in the near future (at least not in the next minor release).

Do you interested help us to make it? Maybe create an RFC or discussion to talk about the solution. ❤️

**Comment by YangKeao** [comment]
If `.spec.targets` are not set for all experiments, they are expected to cooperate with each other. We have some logic to "chain" them together to make sure all of them work. However, if anyone of them have `.spec.targets` set, the situation becomes much more complicated and I didn't think up a way to handle it properly: see [tc_server.go#L139-L161](https://github.com/chaos-mesh/chaos-mesh/blob/master/pkg/chaosdaemon/tc_server.go#L139-L161)

It chains multiple global configurations, and one local config.
